src/storage/exec/IndexNode.h

```diff
@@ -69,6 +69,8 @@ struct InitContext {
   std::vector<std::string> returnColumns;
   // The index of name in `returncolumns`
   Map<std::string, size_t> retColMap;
+  // The columns in statColumns
+  Set<std::string> statColumns;
```

Comment:
I think `statColumns` is redundant. It should be included in `returnColumns`. In my initial design, `returnColumns` contains more than just `property name`, it can be any expression. For example, `max(a)-min(a)`, we need an Aggregate node to calculate max(a) and min(a), and then a Projection is needed to calculate max(a)-min(a) (only subtraction is calculated) At this time, Aggregate should put max(a) and min(a) into `returnColumn` and `retColMap`, and then Projection obtains the values of max(a) and min(a) directly from them.

In this scheme `returnColumns` may require some design. But I think this is the most scalable and least coupled solution. Whether it is `projection {expr}`, `having {expr}`, or `order by {expr}` in the follow-up, the function independence between nodes can be maintained, and the intrusiveness is the lowest.

Reply:
We can remember a TODO first and then merge it first

Reply:
Sounds good! I will add a TODO and fix it in next PR